src/main.rs

```diff
@@ -37,22 +38,50 @@ struct MarketData {
 }
 
 impl MarketData {
-    fn print_holders_map(&self) {
-        for (hash, holders) in &self.files {
+    fn validate_holders(&self, hash: &String) -> Vec<FileRequest> {
+        // make a map of holders already printed
+        let mut previous_holders: HashMap<&String, &FileRequest> = HashMap::new();
+        let mut holders: Vec<FileRequest> = Vec::new();
+
+        for holder in &self.files[hash] {
```

Comment:
The expectation was that only a file hash that exists would be passed in, since it would be run on the registration of a file hash. I can change it to a Some to handle the case where it doesn't exist

Reply:
Got it. Sorry about that; just took a further look of where it's used and `register_file` indeed does ensure that it won't panic since the key would always be there. You can ignore this one then. Thanks!